### PR TITLE
Several compiler warnings and code improvements

### PR DESCRIPTION
**Pull request by garbear**
This PR contains the following compiler warning fixes:

* Misleading code indendation
* Type narrowing conversion

Build system improvements:

* Missing header in CMakeLists.txt

Code improvements:

* Remove several unused functions
* Const-correctness
* Missing `delete` operator
* Missing include

### CODE REVIEW

**Comment by Rechi** [comment]
jenkins build this please

File: xbmc/rendering/gles/RenderSystemGLES.cpp [hunk 1]
```diff
@@ -460,7 +460,7 @@ void CRenderSystemGLES::GetViewPort(CRect& viewPort)
 }
 
 // FIXME make me const so that I can accept temporary objects
-void CRenderSystemGLES::SetViewPort(CRect& viewPort)
+void CRenderSystemGLES::SetViewPort(const CRect& viewPort)
```
Comment (hudokkow) [inline xbmc/rendering/gles/RenderSystemGLES.cpp]: Comment above no longer makes sense?

Comment (garbear) [reply xbmc/rendering/gles/RenderSystemGLES.cpp]: Haha didn't even notice that this completed an 8-year-old FIXME.

Comment (MartijnKaijser) [reply xbmc/rendering/gles/RenderSystemGLES.cpp]: what took you so long?

**Comment by garbear** [comment]
jenkins build this please

**Comment by garbear** [comment]
jenkins build and merge

**Comment by garbear** [comment]
Wayland and Win64 failures unrelated

**Pull request merged**